### PR TITLE
avoid writing and reparsing the artifact usage file once per artifact file in the project deps

### PR DESCRIPTION
**Pull request by KristofferC**
> When in doubt, do things in bulk

The timings below will vary depending on the size of the artifact usage log file but I guess my computer is somewhat representative:

```
julia> ENV["JULIA_PKG_PRECOMPILE_AUTO"] = 0
0

julia> Pkg.activate(; temp=true)

julia> @btime Pkg.add("Plots")
# master
  618.572 ms (4700447 allocations: 339.36 MiB)

# PR
  428.434 ms (3376261 allocations: 268.50 MiB)
``` 

Looking at a TimerOutput profile we can see how the runtime for `download_artifacts` changes:

Master:

```
───────────────────────────────────────────────────────────────────────────────────────────────
                                                      Time                    Allocations      
                                             ───────────────────────   ────────────────────────
              Tot / % measured:                   829ms /  99.4%            340MiB /  99.3%    

Section                              ncalls     time    %tot     avg     alloc    %tot      avg
───────────────────────────────────────────────────────────────────────────────────────────────
add                                       1    825ms  100.0%   825ms    338MiB  100.0%   338MiB
  add                                     1    825ms  100.0%   825ms    338MiB  100.0%   338MiB
...
    download_artifacts                    1    186ms   22.6%   186ms   75.8MiB   22.4%  75.8MiB
...
───────────────────────────────────────────────────────────────────────────────────────────────
```

PR:
```
───────────────────────────────────────────────────────────────────────────────────────────────
                                                      Time                    Allocations      
                                             ───────────────────────   ────────────────────────
              Tot / % measured:                   507ms /  99.6%            269MiB /  99.4%    

Section                              ncalls     time    %tot     avg     alloc    %tot      avg
───────────────────────────────────────────────────────────────────────────────────────────────
add                                       1    505ms  100.0%   505ms    267MiB  100.0%   267MiB
  add                                     1    505ms  100.0%   505ms    267MiB  100.0%   267MiB
...
    download_artifacts                    1   7.46ms    1.3%  7.46ms   7.38MiB    2.7%  7.38MiB
...    
───────────────────────────────────────────────────────────────────────────────────────────────
```

186ms  -> 5.68 ms. 75 MB > 7.38 MB.

### CODE REVIEW

File: src/Types.jl [hunk 1]
```diff
@@ -630,7 +635,9 @@ function write_env_usage(source_file::AbstractString, usage_filepath::AbstractSt
         end
 
         # record new usage
-        usage[source_file] = [Dict("time" => timestamp)]
+        for source_file in source_files
+            usage[source_file] = [Dict("time" => timestamp)]
```
Comment (martinholters) [inline src/Types.jl:639]: As this has been motivated by hunting down unnecessary allocations: Do these have to be seperate objects per `source_file`? (both the single-entry vector and the dict)

Comment (KristofferC) [reply src/Types.jl:639]: Don't think they have to be separate but the cost savings will be extremely low, (7.38MB -> 7.34MB from a quick test). Could maybe do it as an aesthetic thing.

Comment (martinholters) [reply src/Types.jl:639]: Ok, nevermind then. Just wanted to be sure it's a conscious decision, not an oversight.

**Comment by IanButterworth** [comment]
Sounds good. Any benefit of unique!-ing the list too?

**Pull request merged**